src/components/PostDetails.tsx

```diff
@@ -151,14 +99,14 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
             data-cy="WriteCommentButton"
             type="button"
             className="button is-link"
-            onClick={() => setVisible(true)}
+            onClick={toggleCommentForm}
           >
             Write a comment
           </button>
         )}
 
         {loaded && !hasError && visible && (
-          <NewCommentForm onSubmit={addComment} />
+          <NewCommentForm onSubmit={handleAddComment} />
```

Comment:
Verify that the `NewCommentForm` component is correctly implemented and receives the expected props.